### PR TITLE
Added dedicated claim dialect support

### PR DESCRIPTION
**Pull request by supunharsha**

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/twitter/TwitterAuthenticator.java [hunk 1]
```diff
@@ -151,15 +155,17 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
         }
     }
 
-    public void buildClaims(AuthenticationContext context, String jsonObject)
+    private void buildClaims(AuthenticationContext context, String jsonObject)
```
[the remark this answers is not in the document]
Comment (supunharsha) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/twitter/TwitterAuthenticator.java]: The buildClaims method is not called from outside (from authentication framework). It is only used to build calims and the response was sent back by modifying context in "processAuthenticationResponse" method. I have checked this and it works as expected.

[the remark this answers is not in the document]
Comment (hasinthaindrajee) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/twitter/TwitterAuthenticator.java]: What if somebody extended this authentication and wrote his own one ? He might use this. A public API can be used anywhere else. So better to keep it as it is. It's not good to change an api in a minor release or patch release

[the remark this answers is not in the document]
Comment (supunharsha) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/twitter/TwitterAuthenticator.java]: No problem. I can change this as it is. I even agree that we shouldn't do API changes in patch release. But we have inconsistencies in connectors, by making it available. what should we do about the same method in facebook connector which has the private modifier?

[the remark this answers is not in the document]
Comment (supunharsha) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/twitter/TwitterAuthenticator.java]: Sure i will fix it

**Pull request closed without merging**